test/unit-tests/broker-launch.test.js

```diff
@@ -8,23 +8,28 @@ const launchBroker = require('../../dashboard/index');
 // mock event emitter to simulate ChildProcess object
 const mockChildProcess = new EventEmitter();
 
-// sinon fakes for behaviour verification
-let fake = sinon.fake();
-let spawnArgs;
+// sinon spies for behaviour verification
+let unrefSpy = sinon.spy();
+let disconnectSpy = sinon.spy();
+
+// storing stubbed methods arguments for verification
+let spawnArgs, processArgs;
 
 describe('Broker daemonized launching', () => {
     before(() => {
         // stub cp.spawn, to avoid actual daemon launch
         sinon.stub(cp, 'spawn').callsFake((...args) => {
             spawnArgs = args;
 
-            mockChildProcess['unref'] = fake;
-            mockChildProcess['disconnect'] = fake;
+            mockChildProcess['unref'] = unrefSpy;
+            mockChildProcess['disconnect'] = disconnectSpy;
             return mockChildProcess;
         });
 
         // stub process.exit to avoid exiting
-        sinon.stub(process, 'exit');
+        sinon.stub(process, 'exit').callsFake((...args) => {
+            processArgs = args;
+        });
```

Comment:
You don't need to store the reference to args when using stubs.

```suggestion
        sinon.stub(process, 'exit');
```


Later,
```javascript
expect(process.exit.args).to.have.lengthOf(1);
expect(process.exit.args[0]).to.equal(0);
```

Reply:
Okay thank you for this, I didn't know this! I have made the changes!